Framework/HistogramData/src/Interpolate.cpp

```diff
@@ -0,0 +1,209 @@
+#include "MantidHistogramData/Interpolate.h"
+#include "MantidHistogramData/Histogram.h"
+
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_spline.h>
+
+#include <sstream>
+
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::HistogramX;
+using Mantid::HistogramData::HistogramY;
+
+namespace {
+
+constexpr const char *LINEAR_NAME = "Linear";
+constexpr const char *CSPLINE_NAME = "CSpline";
+
+/**
+ * Compute the number of pre-calculated points given the ysize and step size
+ * @param ysize The total number of points in the data set
+ * @param stepSize The step size between each calculated point
+ * @return The number of calculated nodes
+ */
+inline size_t numberCalculated(const size_t ysize, const size_t stepSize) {
```

Comment:
It probably could. The current situations don't know enough at compile time but there is no harm in it being there.